src/shared/varlink.c

```diff
@@ -2871,7 +2871,7 @@ int varlink_get_peer_pidref(Varlink *v, PidRef *ret) {
          * authentication. */
 
         r = varlink_acquire_pidfd(v);
-        if (r < 0)
+        if (r < 0 && !ERRNO_IS_NEG_NOT_SUPPORTED(r))
```

Comment:
Hmm, the handling of v->peer_pidfd is a bit weird? The errno is assigned to that and also returned? It's better to only assign that when >= 0?

Reply:
dunno. i'd say it makes a ton of sense:

we generally say that fds < 0 are invalid. hence whether we leave -EBADF in v->peer_pidfd or some other error shouldn't make a difference. But it's also useful as a cache: you could just look at the variable any time you like and see whith what error it failed again. That's in addition to the error that we immediately return of course, because that's something we always should return.

Reply:
I doubt the error would be reused anywhere. But overall doesn't matter too much.